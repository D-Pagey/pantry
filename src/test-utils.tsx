import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
