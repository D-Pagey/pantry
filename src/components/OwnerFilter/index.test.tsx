import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '../../test-utils';
import { TenantDan, TenantHeidi } from '../../fixtures';
import { OwnerFilter } from '.';

const props = {
    tenants: [TenantDan, TenantHeidi],
    setSelectedTenants: () => null,
    selectedTenants: []
};

describe('OwnerFilter component', () => {
    it('should render', () => {
        const { container } = render(<OwnerFilter {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call setSelectedTenants with selecting a tenant', () => {
        const setSelectedTenants = jest.fn();

        render(<OwnerFilter {...props} setSelectedTenants={setSelectedTenants} />);

        userEvent.click(screen.getByTestId(`ownerFilter${TenantDan.name}`));

        expect(setSelectedTenants).toHaveBeenCalledWith([TenantDan.uid]);
    });

    it('should call setSelectedTenants with correct values when deselecting a tenant', () => {
        const setSelectedTenants = jest.fn();
        const selectedTenants = [TenantDan.uid, TenantHeidi.uid];

        render(<OwnerFilter {...props} selectedTenants={selectedTenants} setSelectedTenants={setSelectedTenants} />);

        // deselect Dan
        userEvent.click(screen.getByTestId(`ownerFilter${TenantDan.name}`));

        expect(setSelectedTenants).toHaveBeenCalledWith([TenantHeidi.uid]);
    });

    it('should call setSelectedTenants with correct values when resetting', () => {
        const setSelectedTenants = jest.fn();
        const selectedTenants = [TenantDan.uid, TenantHeidi.uid];

        render(<OwnerFilter {...props} selectedTenants={selectedTenants} setSelectedTenants={setSelectedTenants} />);

        // deselect Dan
        userEvent.click(screen.getByText('Clear Filter'));

        expect(setSelectedTenants).toHaveBeenCalledWith([]);
    });
});
