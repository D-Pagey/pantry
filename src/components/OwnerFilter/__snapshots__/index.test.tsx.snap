// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerFilter component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-column: 1 / 3;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  list-style: none;
  margin: 0 1rem;
  padding: 0;
}

.c2 {
  border-radius: 50%;
  border: none;
  background: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 0 1rem;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:first-child {
  margin: 0;
}

.c3 {
  border-radius: 50%;
  border: 2px solid white;
  height: 50px;
  width: 50px;
}

.c4 {
  background: none;
  border: none;
  color: #7C7C7C;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:765px) {
  .c0 {
    grid-column: 1 / 2;
    justify-self: start;
  }
}

<div
  class="c0"
>
  <span>
    Filter: 
  </span>
  <ul
    class="c1"
  >
    <button
      class="c2"
      data-testid="ownerFilterDan Page"
      type="button"
    >
      <img
        alt="Dan Page"
        class="c3"
        src="https://lh3.googleusercontent.com/a-/AOh14Gi6ZcKd1ClkJqBEEP114ZJ07XWJfQLKJKL6apgFgQ"
      />
    </button>
    <button
      class="c2"
      data-testid="ownerFilterHeidi Seo"
      type="button"
    >
      <img
        alt="Heidi Seo"
        class="c3"
        src="https://lh3.googleusercontent.com/a-/AOh14Gi6ZcKd1ClkJqBEEP114ZJ07XWJfQLKJKL6apgFgQ"
      />
    </button>
  </ul>
  <button
    class="c4"
    disabled=""
    type="button"
  >
    Clear Filter
  </button>
</div>
`;
